chaco/scaly_plot.py

```diff
@@ -0,0 +1,151 @@
+        return self._make_scale("linear")
+
+    def _log_scale_default(self):
+        return self._make_scale("log")
+
+    def _make_scale(self, scale_type="linear"):
+        """ Returns a new linear or log scale """
+        if scale_type == "linear":
+            if self.linear_scale_factory is not None:
+                return self.linear_scale_factory()
+            else:
+                return ScaleSystem(DefaultScale())
+        else:
+            if self.log_scale_factory is not None:
+                return self.log_scale_factory()
+            else:
+                return ScaleSystem(LogScale())
+
+    def _init_components(self):
+        # Since this is called after the HasTraits constructor, we have to make
+        # sure that we don't blow away any components that the caller may have
+        # already set.
+
+        if self.range2d is None:
+            self.range2d = DataRange2D()
+
+        if self.index_mapper is None:
+            if self.index_scale == "linear":
+                imap = LinearMapper(range=self.range2d.x_range)
+            else:
+                imap = LogMapper(range=self.range2d.x_range)
+            self.index_mapper = imap
+
+        if self.value_mapper is None:
+            if self.value_scale == "linear":
+                vmap = LinearMapper(range=self.range2d.y_range)
+            else:
+                vmap = LogMapper(range=self.range2d.y_range)
+            self.value_mapper = vmap
+
+        if self.x_ticks is None:
+            self.x_ticks = ScalesTickGenerator(
+                scale=self._make_scale(self.index_scale)
+            )
+        if self.y_ticks is None:
+            self.y_ticks = ScalesTickGenerator(
+                scale=self._make_scale(self.value_scale)
+            )
+
+        if self.x_grid is None:
+            self.x_grid = PlotGrid(
+                mapper=self.x_mapper,
+                orientation="vertical",
+                line_color="lightgray",
+                line_style="dot",
+                component=self,
+                tick_generator=self.x_ticks,
+            )
+        if self.y_grid is None:
+            self.y_grid = PlotGrid(
+                mapper=self.y_mapper,
+                orientation="horizontal",
+                line_color="lightgray",
+                line_style="dot",
+                component=self,
+                tick_generator=self.y_ticks,
+            )
+        if self.x_axis is None:
+            self.x_axis = PlotAxis(
+                mapper=self.x_mapper,
+                orientation="bottom",
+                component=self,
+                tick_generator=self.x_ticks,
+            )
+        if self.y_axis is None:
+            self.y_axis = PlotAxis(
+                mapper=self.y_mapper,
+                orientation="left",
+                component=self,
+                tick_generator=self.y_ticks,
+            )
+
+    def _index_scale_changed(self, old, new):
+        Plot._index_scale_changed(self, old, new)
```

Comment:
Not entirely sure how this works. Shouldn't this be `super()._index_scale_changed(old, new)`? Same comment below with `Plot._value_scale_changed(...)`

Reply:
I agree, super makes more sense / is cleaner here.  This code I believe is also untested, I simply copy pasted the old file to revive it.
It could probably use a closer look / some updates / some new tests.  I can do so in some follow up PRs